### PR TITLE
Updates Packer provisioning guide with HCL2

### PR DESCRIPTION
**Pull request by tr0njavolta**

### CODE REVIEW

File: images/image.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,36 @@
+variable "region" {
+  type    = string
+  default = "us-east-1"
+}
+
+# source blocks are generated from your builders; a source can be referenced in
+# build blocks. A build block runs provisioners and post-processors on a
+# source.
+source "amazon-ebs" "example" {
+  ami_name      = "learn-terraform-packer"
```
Comment (topfunky) [inline images/image.pkr.hcl]:
```suggestion
  ami_name      = "learn-terraform-packer-{{timestamp}}"
```

This fails with an error. The JSON version used a timestamp in the name in order to generate a (mostly) unique name. Maybe this needs `{{timestamp}}` too?

```
==> amazon-ebs.example: Error: AMI Name: 'learn-terraform-packer' is used by an existing AMI: ami-0dbaca5d269497603
Build 'amazon-ebs.example' errored after 1 second 791 milliseconds: Error: AMI Name: 'learn-terraform-packer' is used by an existing AMI: ami-0dbaca5d269497603
```

**Review by topfunky — CHANGES_REQUESTED** [review]
I ran into some errors. Are these unique to my machine or do you see the same error?

```
==> amazon-ebs.example: + sudo apt-get -y -qq install curl wget git vim apt-transport-https ca-certificates
==> amazon-ebs.example: + sudo add-apt-repository ppa:longsleep/golang-backports -y
==> amazon-ebs.example: gpg: keyring `/tmp/tmpp2gx897l/secring.gpg' created
==> amazon-ebs.example: gpg: keyring `/tmp/tmpp2gx897l/pubring.gpg' created
==> amazon-ebs.example: gpg: requesting key 56A3D45E from hkp server keyserver.ubuntu.com
==> amazon-ebs.example: gpg: /tmp/tmpp2gx897l/trustdb.gpg: trustdb created
==> amazon-ebs.example: gpg: key 56A3D45E: public key "Launchpad PPA for Simon Eisenmann" imported
==> amazon-ebs.example: gpg: Total number processed: 1
==> amazon-ebs.example: gpg:               imported: 1  (RSA: 1)
    amazon-ebs.example: OK
==> amazon-ebs.example: + sudo apt-get -y -qq install golang-go
==> amazon-ebs.example: E: Could not open file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_universe_binary-amd64_Packages - open (2: No such file or directory)
==> amazon-ebs.example: E: Could not open file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_restricted_i18n_Translation-en - open (2: No such file or directory)
==> amazon-ebs.example: E: Could not open file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_restricted_binary-amd64_Packages - open (2: No such file or directory)
==> amazon-ebs.example: E: Could not open file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_main_i18n_Translation-en - open (2: No such file or directory)
==> amazon-ebs.example: E: Could not open file /var/lib/apt/lists/archive.ubuntu.com_ubuntu_dists_xenial_main_binary-amd64_Packages - open (2: No such file or directory)
==> amazon-ebs.example: + sudo groupadd -r hashicorp
==> amazon-ebs.example: + sudo useradd -m -s /bin/bash terraform
==> amazon-ebs.example: + sudo usermod -a -G hashicorp terraform
==> amazon-ebs.example: + sudo cp /etc/sudoers /etc/sudoers.orig
==> amazon-ebs.example: + sudo tee /etc/sudoers.d/terraform
==> amazon-ebs.example: + echo 'terraform  ALL=(ALL) NOPASSWD:ALL'
    amazon-ebs.example: terraform  ALL=(ALL) NOPASSWD:ALL
==> amazon-ebs.example: + sudo mkdir -p /home/terraform/.ssh
==> amazon-ebs.example: + sudo chmod 700 /home/terraform/.ssh
==> amazon-ebs.example: + sudo cp /tmp/tf-packer.pub /home/terraform/.ssh/authorized_keys
==> amazon-ebs.example: + sudo chmod 600 /home/terraform/.ssh/authorized_keys
==> amazon-ebs.example: + sudo chown -R terraform /home/terraform/.ssh
==> amazon-ebs.example: + sudo usermod --shell /bin/bash terraform
==> amazon-ebs.example: usermod: no changes
==> amazon-ebs.example: + sudo -H -i -u terraform -- env bash
    amazon-ebs.example: terraform
    amazon-ebs.example: /home/terraform
==> amazon-ebs.example: bash: line 9: go: command not found
==> amazon-ebs.example: Provisioning step had errors: Running the cleanup provisioner, if present...
==> amazon-ebs.example: Terminating the source AWS instance...
==> amazon-ebs.example: Cleaning up any extra volumes...
==> amazon-ebs.example: No volumes to clean up, skipping
==> amazon-ebs.example: Deleting temporary security group...
==> amazon-ebs.example: Deleting temporary keypair...
Build 'amazon-ebs.example' errored after 1 minute 57 seconds: Script exited with non-zero exit status: 127.Allowed exit codes are: [0]
```

File: images/image.pkr.hcl (hunk 1 above)
Comment (tr0njavolta) [inline images/image.pkr.hcl]:
```suggestion
"learn-terraform-packer-${formatdate("MMDDYYYY-HHmmss", timestamp())}"
```

Comment (tr0njavolta) [reply images/image.pkr.hcl]: Updated with timestamp from tutorial text.

Comment (SwampDragons) [reply images/image.pkr.hcl]: "timestamp" won't work this way in HCL. You'd need to create a local var, above the source block: 

`locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }`

and access it like this: 

`"learn-terraform-packer-${local.timestamp}"`

Comment (tr0njavolta) [reply images/image.pkr.hcl]: Thanks! Just tested and updated.

**Review by im2nguyen — APPROVED** [review]

**Review by topfunky — APPROVED** [review]

**Pull request merged**